src/multiinstances/internal/ipc/ipcsocket.h

```diff
@@ -0,0 +1,70 @@
+/*
+ * SPDX-License-Identifier: GPL-3.0-only
+ * MuseScore-CLA-applies
+ *
+ * MuseScore
+ * Music Composition & Notation
+ *
+ * Copyright (C) 2021 MuseScore BVBA and others
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+#ifndef MU_IPC_IPCSOCKET_H
+#define MU_IPC_IPCSOCKET_H
+
+#include "ipc.h"
+
+#include "async/channel.h"
+#include "async/notification.h"
+#include "async/asyncable.h"
+
+class QLocalSocket;
+
+namespace mu::ipc {
+class IpcLock;
+class IpcSocket : public async::Asyncable
+{
+public:
+    IpcSocket() = default;
+    ~IpcSocket();
+
+    const ID& selfID() const;
+
+    bool connect(const QString& serverName);
+    async::Notification disconected();
+
+    bool send(const Msg& msg);
+    async::Channel<Msg> msgReceived() const;
+
+    QList<Meta> instances() const;
+    async::Notification instancesChanged() const;
+
+private:
+
+    bool doSendToSocket(QLocalSocket* socket, const QByteArray& data);
```

Comment:
I thought that IpcServer is the guy who is responsible for that, could you explain why would we need this method in socket?

Reply:
It's just for historical reasons ...
At first, server and socket were one class ...
This one changed many times, and at some moments it was difficult, I didn't want to duplicate it ...
I also wanted the server and the socket to have this method, at that moment, so that it was easy to find this common place, especially at a time when it was difficult to implement.
Most of the reasons are not relevant now, so I can change it.

Reply:
removed this method